src/parachain/vaults.ts

```diff
@@ -541,20 +547,10 @@ export class DefaultVaultsAPI extends DefaultTransactionAPI implements VaultsAPI
     }
 
     async getTotalIssuableAmount(): Promise<MonetaryAmount<WrappedCurrency, BitcoinUnit>> {
-        const perCollateralIssuableAmounts = await Promise.all(
-            CollateralCurrency.map(
-                (currency) =>
-                    new Promise<MonetaryAmount<WrappedCurrency, BitcoinUnit>>((resolve) => {
-                        this.tokensAPI.total(currency as Currency<CurrencyUnit>).then(async (nomination) => {
-                            const capacity = await this.calculateCapacity(
-                                nomination as MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>
-                            );
-                            resolve(capacity);
-                        });
-                    })
-            )
+        const perVaultIssuableAmounts = await Promise.all(
```

Comment:
This query is very inefficient and has degrades with the number of vaults since you need to query each vault from the parachain node and then loop through all the vaults. Instead, use the TotalUserVaultCollateral for all collateral currencies defined here https://github.com/interlay/interbtc/blob/master/crates/vault-registry/src/lib.rs#L633 plus the total issued wrapped tokens at the moment to get the total issuable amount.